merlin-server/constraints-dsl-compiler/src/libs/constraints-edsl-fluent-api.ts

```diff
@@ -323,6 +323,26 @@ export class Windows {
       defaultProfile: defaultProfile.__astNode
     });
   }
+
+  /**
+   * Counts the cumulative amount of time spent in an active Span, as a multiple of some unit of time.
+   *
+   * WARNING: This method does not guarantee any particular zero point. Meaning, don't base behavior off of
+   * the total absolute time over the whole plan, because features such as temporal subset branches or
+   * simulation might change the zero point.
+   *
+   * Instead, use this for *relative* changes. Unfortunately, this requires subtraction, which is
+   * surprisingly difficult right now. We're working on it.
```

Comment:
Can you detail this here in the discussion ? I am not grasping what you mean to explain. Maybe give an example ?

Reply:
Mathematically this should be treated like an [affine space](https://en.wikipedia.org/wiki/Affine_space). A good example from physics is the energy contained in a system. In most situations except those dealing with absolute zero, there isn't a universal choice of what should be labelled as "zero energy", because different observers can disagree on what would be the most natural zero point for their calculations. But all observers will ultimately agree on how the system behaves because its only differences/changes in energy that matter to the evolution. In electronics, the decision that ground is zero volts is an arbitrary convention. It would be just as valid to declare that ground is 5 volts and all the predictions would be the same, although the math would be a little more annoying.

In our case, the duration profile will always start at 0 at the beginning of the simulation data, so it seems like you could write a constraint that says "don't do more than 1 hour of activity A in this plan". But if we had plan subset branches, the start time of simulation might change, some activities might be omitted, and `accumulatedDuration` will assign a different point the honor of being `0`. So its a lot more error-prone to reason about total time, and a lot more reliable to reason about changes/differences in time.

There's also a typo in that comment, thanks for making be re-read it :)

Reply:
Can we also just plaster your amazing explanation all over the simulation engine, please? Thanks 😂